skimage/filters/rank/generic.py

```diff
@@ -64,16 +63,54 @@
 
 
 def _handle_input(image, selem, out, mask, out_dtype=None, pixel_size=1):
+    """Preprocess and verify input for filters.rank methods.
 
+    Parameters
+    ----------
+    image : 2-D array (integer, float or boolean)
+        Input image.
+    selem : 2-D array (integer, float or boolean)
+        The neighborhood expressed as a 2-D array of 1's and 0's.
+    out : 2-D array (integer, float or boolean)
+        If None, a new array is allocated.
+    mask : ndarray (integer, float or boolean)
+        Mask array that defines (>0) area of the image included in the local
+        neighborhood. If None, the complete image is used (default).
+    out_dtype : data-type
+        Desired output data-type. Default is None, which means we cast output
+        in input dtype.
+    pixel_size : int
+        Dimension of each pixel. Default value is 1.
+
+    Returns
+    -------
+    image : 2-D array (np.uint8 or np.uint16)
+    selem : 2-D array (np.uint8)
+        The neighborhood expressed as a binary 2-D array.
+    out : 3-D array (same dtype out_dtype or as input)
+        Output array. The two first dimensions are the spatial ones, the third
+        one is the pixel vector (length 1 by default).
+    mask : 2-D array (np.uint8)
+        Mask array that defines (>0) area of the image included in the local
+        neighborhood.
+    n_bins : int
+        Number of histogram bins.
+    out_dtype : data-type
+        Output data-type.
+
+    """
     assert_nD(image, 2)
-    if image.dtype not in (np.uint8, np.uint16):
+    if image.dtype not in (np.uint8, np.uint16, np.bool_):
         message = ('Possible precision loss converting image of type {} to '
                    'uint8 as required by rank filters. Convert manually using '
                    'skimage.util.img_as_ubyte to silence this warning.'
                    .format(image.dtype))
         warn(message, stacklevel=2)
```

Comment:
```suggestion
        warn(message, stacklevel=3)
```

Reply:
Actually, it could be as high as 5!

Reply:
To get the warnings at the correct level, you also need to remvoe the `default_selem` decorator.

```python
if selem is None:
        selem = ndi.generate_binary_structure(image.ndim, image.ndim)
```
and just that line where the decorator used to be used.

Reply:
When you say remove the decorator, do you mean to fully delete it (as in deleting the entire `default_selem` method) and Line 602 (`@default_elem`) as the line where the decorator used to be used?

Reply:
yes, because the decorator adds a "stack level" that you would have to indicate in the function.